Program.cs

```diff
@@ -0,0 +1,238 @@
+    }
+
+    
+    public static RestResponse Login(RestClient client, string email, string password) 
+    {
+        
+        RestRequest request = new("/authaccount/login", Method.Post);
+        
+        Dictionary<string, string> payload = new()
+        {
+            {"email", email},
+            {"password", password}
+        };
+        var jsonReq = JsonConvert.SerializeObject(payload);
+        
+        request.AddJsonBody(jsonReq);
+        
+        RestResponse response = client.Execute(request);
+        return response;
+    }
+
+    public static RestResponse Register(RestClient client, string email, string password, string name)
+    {
+        RestRequest request = new("/authaccount/registration", Method.Post);
+        
+        Dictionary<string, string> payload = new()
+        {
+            {"email", email},
+            {"password", password},
+            {"name", name}
+        };
+        var jsonReq = JsonConvert.SerializeObject(payload);
+        
+        request.AddJsonBody(jsonReq);
+        
+        RestResponse response = client.Execute(request);
+        return response;
+    }
+    
+    
+    public static void CreateTable() 
+    {        
+        using SQLiteConnection connection = new($"Data Source=users.db;Version=3;");
+        connection.Open();   
+        
+        using SQLiteCommand command = new("CREATE TABLE IF NOT EXISTS Users (Id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, email TEXT, createdAt DATE)", connection);
+        command.ExecuteNonQuery();
+        
+        connection.Close();
+
+    }
+
+    // name, email, createdAt
+    public static void InsertInto(User? user) 
+    {
+        if (user == null) {
+            Console.WriteLine("Failed to retrive user data");
+            return;
+        }
+        
+        using SQLiteConnection connection = new($"Data Source=users.db;Version=3;");
+        connection.Open();   
+        
+        using SQLiteCommand command = new("INSERT INTO Users (name, email, createdAt) VALUES (@Name, @Email, DATE())", connection);
+        command.Parameters.AddWithValue("@Name", user.Name);
+        command.Parameters.AddWithValue("@Email", user.Email);
+        command.ExecuteNonQuery();
+        
+        command.ExecuteNonQuery();
+        
+        connection.Close();
+    }
+
+    public static void ExportUsers() 
+    {
+        using SQLiteConnection connection = new($"Data Source=users.db;Version=3;");
+        connection.Open();   
+        
+        using SQLiteCommand command = new("SELECT * FROM Users", connection);
+        var reader = command.ExecuteReader();
+        using StreamWriter outputFile = new StreamWriter("./export.csv");
+        while (reader.Read()) {
+            var csvLine = $"{reader["id"]},{reader["name"]},{reader["email"]},{reader["createdAt"]}";
+            outputFile.WriteLine(csvLine);
+        }
+        connection.Close();
+    }
+
+    public static void CrawlUsers(RestClient client)
```

Comment:
For this function, I would have liked to have made the function take two more parameters: totalPages and startingPage

Then I could have leveraged an threadpool with batches of 4 to leverage multiprocessing to be more comprehensive. I tried to initialize total pages based on the first result to be equal to the total number of pages, but it was taking too long.